src/common_defaults.jl

```diff
@@ -27,6 +27,7 @@ Base.mapreduce_empty(::typeof(UNITLESS_ABS2), op, T) = abs2(Base.reduce_empty(op
 @inline NAN_CHECK(x::Enum) = false
 @inline NAN_CHECK(x::AbstractArray) = any(NAN_CHECK, x)
 @inline NAN_CHECK(x::RecursiveArrayTools.ArrayPartition) = any(NAN_CHECK, x.x)
+@inline NAN_CHECK(x::DEDataArray) = any(NAN_CHECK, x.x)

```

Comment:
I guess it would be better to use
```suggestion
@inline NAN_CHECK(x::DEDataArray) = NAN_CHECK(x.x)

```
since then it will specialize on the type of `x.x` (which usually calls `any(NAN_CHECK, x)` apparently but there might be special implementations for specific types).

Reply:
ArrayPartitions of ArrayPartitions would be a case this accelerates.

Reply:
Oh I see, that makes sense! I'll change it.